common/djangoapps/student/models.py

```diff
@@ -1741,6 +1745,11 @@ def dynamic_upgrade_deadline(self):
 
         course_config = CourseDynamicUpgradeDeadlineConfiguration.current(self.course_id)
         if course_config.enabled and course_config.opt_out:
+            # Course-level config should be checked first since it overrides the org-level config
+            return None
+
+        org_config = OrgDynamicUpgradeDeadlineConfiguration.current(self.course_id.org)
+        if org_config.enabled and org_config.opt_out and not (course_config.enabled and not course_config.opt_out):
```

Comment:
Nit: If we're always going to be calling these things in combination like `course_config.enabled and not course_config.opt_out`, it might be better to consolidate that into one or two helper functions within the config class definitions, e.g.:

```python
if org_config.dont_send() and course_config.dont_send():
```

Reply:
Agreed. I'm having trouble groking the double negatives here.

Reply:
I added `opted_in()` and `opted_out()` to check for: enabled=True & opt_out=False and enabled=True & opt_out=True respectively.